File: QuickSort.java

public class QuickSort {

    // Method to perform Quick Sort
    public static void quickSort(int[] array, int low, int high) {

        // Check if there is more than one element
        if (low < high) {

            // Find the partition index
            int partitionIndex = partition(array, low, high);

            // Sort the left part
            quickSort(array, low, partitionIndex - 1);

            // Sort the right part
            quickSort(array, partitionIndex + 1, high);

        }

    }

    // Method to partition the array
    public static int partition(int[] array, int low, int high) {

        // Choose the last element as pivot
        int pivot = array[high];

        // Index of smaller element
        int i = low - 1;

        // Compare each element with the pivot
        for (int j = low; j < high; j++) {

            // If current element is smaller than or equal to pivot
            if (array[j] <= pivot) {

                i++;

                // Swap array[i] and array[j]
                int temp = array[i];
                array[i] = array[j];
                array[j] = temp;

            }

        }

        // Place pivot in its correct position
        int temp = array[i + 1];
        array[i + 1] = array[high];
        array[high] = temp;

        // Return pivot index
        return i + 1;
    }

    // Method to print the array
    public static void printArray(int[] array) {

        for (int number : array) {
            System.out.print(number + " ");
        }

        System.out.println();
    }

    public static void main(String[] args) {

        // Unsorted array
        int[] numbers = {10, 7, 8, 9, 1, 5};

        System.out.println("Before Sorting:");
        printArray(numbers);

        // Call Quick Sort
        quickSort(numbers, 0, numbers.length - 1);

        System.out.println("After Sorting:");
        printArray(numbers);

    }
}

File: QuickSort.c
#include <stdio.h>

// Function to swap two elements
void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

// Function to partition the array
int partition(int arr[], int low, int high) {

    // Select the last element as pivot
    int pivot = arr[high];

    int i = low - 1;

    for(int j = low; j < high; j++) {

        if(arr[j] < pivot) {
            i++;
            swap(&arr[i], &arr[j]);
        }

    }

    swap(&arr[i + 1], &arr[high]);

    return i + 1;
}

// Quick Sort function
void quickSort(int arr[], int low, int high) {

    if(low < high) {

        int pivotIndex = partition(arr, low, high);

        // Sort left part
        quickSort(arr, low, pivotIndex - 1);

        // Sort right part
        quickSort(arr, pivotIndex + 1, high);
    }
}

int main() {

    int arr[100];
    int n, i;

    printf("Enter the number of elements: ");
    scanf("%d", &n);

    printf("Enter %d elements:\n", n);

    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    quickSort(arr, 0, n - 1);

    printf("\nSorted Array (Ascending Order):\n");

    for(i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    printf("\n");

    return 0;
}
